R/sub.R

```diff
@@ -64,11 +64,12 @@
 str_sub <- function(string, start = 1L, end = -1L) {
   vctrs::vec_size_common(string = string, start = start, end = end)
 
-  if (is.matrix(start)) {
+  out <- if (is.matrix(start)) {
     stri_sub(string, from = start)
   } else {
     stri_sub(string, from = start, to = end)
   }
+  if (length(out) == length(string)) copy_names(string, out) else out
```

Comment:
Isn't `length(out) == length(string)` always going to be true?

Reply:
We could add `# Preserve names unless `string` is recycled` before the final line of `str_sub()` and likewise for `str_pad()`. But I see that the branch has now been merged. Would you still like me to (pull and) commit?

Reply:
I just merged the updated main branch into your branch since I made a bunch of merge conflicts. You  can continue to edit it.

Reply:
@hadley, I added comments in 669a0e7 as described. Let me know if there is anything missing. Thanks!